packages/video_player/lib/video_player.dart

```diff
@@ -428,50 +398,107 @@ class _VideoProgressBarState extends State<VideoProgressBar> {
   }
 }
 
-class ProgressBarPainter extends CustomPainter {
-  VideoPlayerValue value;
-  VideoProgressColors colors;
-  ProgressBarPainter(this.value, this.colors);
+/// Displays the play/buffering status of the video controlled by [controller].
+///
+/// If [allowScrubbing] is true, this widget will detect taps and drags and
+/// seek the video accordingly.
+///
+/// [padding] allows to specify some extra padding around the progress indicator
+/// that will also detect the gestures.
+class VideoProgressIndicator extends StatefulWidget {
+  final VideoPlayerController controller;
+  final VideoProgressColors colors;
+  final bool allowScrubbing;
+  final EdgeInsets padding;
+
+  VideoProgressIndicator(this.controller,
+      {VideoProgressColors colors,
+      this.allowScrubbing,
+      this.padding: const EdgeInsets.only(top: 5.0)})
+      : colors = colors ?? new VideoProgressColors();
+
+  @override
+  _VideoProgressIndicatorState createState() {
+    return new _VideoProgressIndicatorState();
```

Comment:
`=>` syntax may be preferable here, depending on whether you get a line break.

Reply:
done